packages/wallet-sdk/src/core/provider/interface.ts

```diff
@@ -42,10 +42,47 @@ export interface AppMetadata {
   appChainIds: number[];
 }
 
-export interface Preference {
+type PostOnboardingAction = 'none' | 'onramp' | 'magicspend';
+
+type OnrampPrefillOptions = {
+  contractAddress?: string;
+  amount: string;
+  chainId: number;
+};
+
+export type Preference = {
   options: 'all' | 'smartWalletOnly' | 'eoaOnly';
+  /**
+   * @deprecated this is only for internal use
+   */
   keysUrl?: string;
-}
+  /**
+   * @param postOnboardingAction
+   * @type {PostOnboardingAction}
+   * @description Recommends the action to be taken after the onboarding process to engage users with various flows. Possible values:
+   * - `none`: No action is recommended post-onboarding. (Default experience)
+   * - `onramp`: Recommends initiating the onramp flow, allowing users to prefill their account with a specified asset, chain, and amount.
```

Comment:
> allowing users to prefill their account with a specified asset, chain, and amount.

"allowing users to prefill their account with an optional <OnrampPrefillOptions>"